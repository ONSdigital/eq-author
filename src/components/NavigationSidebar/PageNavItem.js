import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { buildQuestionnairePath } from "utils/UrlUtils";
import NavLink from "./NavLink";
import PageIcon from "./icon-questionpage.svg?inline";
import { withRouter } from "react-router-dom";
import CustomPropTypes from "custom-prop-types";
import gql from "graphql-tag";

import { size } from "lodash";
import { compose } from "redux";
import { connect } from "react-redux";

const StyledPageItem = styled.li`
  padding: 0;
  margin: 0;
  position: relative;
  display: flex;
  align-items: center;
`;

export const UnwrappedPageNavItem = ({
  sectionId,
  questionnaireId,
  page,
  match,
  errors,
  ...otherProps
}) => (
  <StyledPageItem data-test="page-item" {...otherProps}>
    <NavLink
      to={buildQuestionnairePath({
        questionnaireId,
        sectionId,
        pageId: page.id,
        tab: match.params.tab || "design"
      })}
      title={page.displayName}
      icon={PageIcon}
      errors={errors}
      data-test="nav-page-link"
    >
      {page.displayName}
    </NavLink>
  </StyledPageItem>
);

UnwrappedPageNavItem.fragments = {
  PageNavItem: gql`
    fragment PageNavItem on QuestionPage {
      id
      title
      position
      displayName
    }
  `
};

UnwrappedPageNavItem.propTypes = {
  sectionId: PropTypes.string.isRequired,
  questionnaireId: PropTypes.string.isRequired,
  page: CustomPropTypes.page,
  match: CustomPropTypes.match
};

const mapStateToProps = (state, ownProps) => ({
  errors: size(state.fieldValidation.errors[ownProps.page.id])
});

export default compose(
  withRouter,
  connect(mapStateToProps)
)(UnwrappedPageNavItem);
