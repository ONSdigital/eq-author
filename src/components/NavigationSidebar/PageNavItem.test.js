import React from "react";
import { shallow } from "enzyme";
import PageNavItem, { LinkText } from "./PageNavItem";

describe("PageNavItem", () => {
  let component;

  beforeEach(() => {
    component = shallow(
      <PageNavItem
        questionnaireId={"1"}
        sectionId={"2"}
        pageId={"3"}
        title="Title"
      />
    );
  });

  it("should render", () => {
    expect(component).toMatchSnapshot();
  });

  describe("LinkText", () => {
    it("should be opaque", () => {
      const wrapper = shallow(<LinkText />);
      expect(wrapper).toMatchSnapshot();
    });

    it("should be partially transparent if fade prop passed", () => {
      const wrapper = shallow(<LinkText fade />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
