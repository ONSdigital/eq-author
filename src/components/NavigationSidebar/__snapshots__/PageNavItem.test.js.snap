// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNavItem LinkText should be opaque 1`] = `
.c0 {
  display: inline-block;
  vertical-align: middle;
  width: 100%;
  position: relative;
  z-index: 2;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  opacity: 1;
  font-size: 0.75em;
}

<span
  className="c0"
/>
`;

exports[`PageNavItem LinkText should be partially transparent if fade prop passed 1`] = `
.c0 {
  display: inline-block;
  vertical-align: middle;
  width: 100%;
  position: relative;
  z-index: 2;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  opacity: 0.5;
  font-size: 0.75em;
}

<span
  className="c0"
/>
`;

exports[`PageNavItem should render 1`] = `
<PageNavItem__StyledPageItem
  data-test="page-item"
>
  <PageNavItem__Link
    activeClassName="selected"
    aria-disabled={false}
    to="/questionnaire/1/design/2/3"
  >
    <PageNavItem__LinkText
      fade={false}
    >
      Title
    </PageNavItem__LinkText>
  </PageNavItem__Link>
</PageNavItem__StyledPageItem>
`;
